Automattic/Gravatar-SDK-Android
# Presentation logic for the OAuthPage

### Description
Another Draft PR to get some initial feedback. 

We haven't yet needed to do more complicated presentation logic so we don't have any base standards for that. 

This PR adds the ViewModel for the `OAuthPage`. 

I've decided to do manual DI to not add any more dependencies for now. 
One problem is that the `ViewModel` is scoped to the Activity so in our case we would always get the same `ViewModel`. To overcome that I've added the `navigation-compose` because each destination acts as a `ViewModelStoreOwner`. 

This allowed me to move some logic out of the composable like the `launchCustomTab()` call that was triggered in `DisposableEffect`.

One thing I had to add because of the `navigation-compose` is the `GravatarQuickEditorSplashPage`. It acts as the start destination in the nav graph and will handle the routing later when we store the token. There needs to be a `startDestination` defined so we need something and I don't want to start with the `oauth` route and close it quickly if we detect that the token is present. 

For the `ViewModel` itself, there's nothing extraordinary. One thing different than what we can be used to is that the `Actions`  emitted from `ViewModel` are separate and not part of the `UiState`. 


### Testing Steps

Add these to your local.properties

```
demo-app.wordpress.oauth.clientId=yourId
demo-app.wordpress.oauth.clientSecret=yourSecret
demo-app.wordpress.oauth.redirectUri=yourURI
```
and play with the demo app.


1. Tap the `UpdateAvatar`
2. Continue with the required steps to finish the OAuthFlow
3. `Insert the real avatar picker page` text should appear - this is a placeholder for the avatar picker UI
4. Tap the above text
5. Toast message should appear with a confirmation
6. Tap the `UpdateAvatar` 
7. Close the browser with the `x` button
8. Tap the `Done` button
9. Toast with `Dissmised` info is shown
10. Tap the `UpdateAvatar` 
11. Tap `Deny`
12. Toast with `Error` info is shown
